ddl/ddl_worker.go

```diff
@@ -286,14 +290,23 @@ func (d *ddl) handleDDLJobQueue(shouldCleanJobs bool) error {
 
 			// If running job meets error, we will save this error in job Error
 			// and retry later if the job is not cancelled.
-			schemaVer, err = d.runDDLJob(t, job)
+			schemaVer, runJobErr = d.runDDLJob(t, job)
 			if job.IsCancelled() {
 				err = d.finishDDLJob(t, job)
 				return errors.Trace(err)
 			}
-			err = d.updateDDLJob(t, job, err != nil)
+			err = d.updateDDLJob(t, job, runJobErr != nil)
 			return errors.Trace(d.handleUpdateJobError(t, job, err))
 		})
+
+		if runJobErr != nil {
+			// wait a while to retry again. If we don't wait here, ddl will retry this job immediately,
```

Comment:
s/ddl/DDL

Reply:
It's a cherry-pick, we should modify it in a new PR